examples/streaming/knative-eventing/README.md

```diff
@@ -0,0 +1,387 @@
+  name: simple-iris-spec
+  predictors:
+  - graph:
+      implementation: SKLEARN_SERVER
+      modelUri: gs://seldon-models/sklearn/iris
+      name: simple-iris-model
+      children: []
+    name: simple-iris-predictor
+    replicas: 1
+
+```
+
+    Overwriting ./assets/simple-iris-deployment.yaml
+
+
+### Run the model in our cluster
+
+Now we run the Seldon Deployment configuration file we just created.
+
+
+```python
+!kubectl apply -f assets/simple-iris-deployment.yaml
+```
+
+    seldondeployment.machinelearning.seldon.io/simple-iris-deployment created
+
+
+### Check that the model has been deployed
+
+
+
+
+```python
+!kubectl get pods
+```
+
+    NAME                                                              READY   STATUS    RESTARTS   AGE
+    default-broker-filter-7ffddb5dcc-pdbdt                            1/1     Running   1          7d8h
+    default-broker-ingress-5cfc4c8cbc-dqr82                           1/1     Running   1          7d8h
+    seldon-controller-manager-5f8cfb4648-jfltq                        1/1     Running   0          163m
+    simple-iris-deployment-simple-iris-predictor-0-6dd54c578c-6gc6p   2/2     Running   0          6m58s
+
+
+## Create a Trigger to reach our model 
+
+We want to create a trigger that is able to reach directly to the service.
+
+We will be using the following seldon deployment:
+
+
+```python
+!kubectl get sdep
+```
+
+    NAME                     AGE
+    simple-iris-deployment   3m22s
+
+
+### Create trigger configuration
+
+
+```python
+%%writefile ./assets/seldon-knative-trigger.yaml
+
+apiVersion: eventing.knative.dev/v1beta1
+kind: Trigger
+metadata:
+  name: seldon-eventing-sklearn-trigger
+spec:
+  broker: default
+  filter:
+    attributes:
+      type: seldon.simple-iris-deployment.request
+  subscriber:
+    ref: 
+      apiVersion: machinelearning.seldon.io/v1
+      kind: SeldonDeployment
+      name: simple-iris-deployment
+
+```
+
+    Overwriting ./assets/seldon-knative-trigger.yaml
+
+
+Create this trigger file which will send all cloudevents of type `"seldon.<deploymentName>.request"`.
+
+
+```python
+!kubectl apply -f assets/seldon-knative-trigger.yaml
+```
+
+    trigger.eventing.knative.dev/seldon-eventing-sklearn-trigger unchanged
+
+
+CHeck that the trigger is working correctly (you should see "Ready: True"), together with the URL that will be reached.
+
+
+```python
+!kubectl get trigger 
+```
+
+    NAME                              READY   REASON   BROKER    SUBSCRIBER_URI                                                                                                          AGE
+    event-display                     True             default   http://event-display.default.svc.cluster.local/                                                                         3d5h
+    seldon-eventing-sklearn-trigger   True             default   http://istio-ingressgateway.istio-system.svc.cluster.local/seldon/default/simple-iris-deployment/api/v1.0/predictions   74m

```

Comment:
How does the `Trigger` know which URL it needs to access on the `SeldonDeployment`? Is that through the `Addressable` field on the status?

Reply:
That is correct! We are using the concept that they refer to as "duck typing" which basically allows you to reference an API through its kubernetes resource, and it would be able to infer what the endpoint is through this addressable URL. (You can also still override it on the trigger)